src/Umbraco.Web.BackOffice/Controllers/MemberGroupController.cs

```diff
@@ -26,45 +28,62 @@ public class MemberGroupController : UmbracoAuthorizedJsonController
         private readonly IMemberGroupService _memberGroupService;
         private readonly UmbracoMapper _umbracoMapper;
         private readonly ILocalizedTextService _localizedTextService;
+        private readonly RoleManager<IdentityRole> _roleManager;
 
         public MemberGroupController(
             IMemberGroupService memberGroupService,
             UmbracoMapper umbracoMapper,
-            ILocalizedTextService localizedTextService
+            ILocalizedTextService localizedTextService,
+            RoleManager<IdentityRole> roleManager
             )
         {
             _memberGroupService = memberGroupService ?? throw new ArgumentNullException(nameof(memberGroupService));
+            _roleManager = roleManager ?? throw new ArgumentNullException(nameof(roleManager));
             _umbracoMapper = umbracoMapper ?? throw new ArgumentNullException(nameof(umbracoMapper));
-            _localizedTextService =
-                localizedTextService ?? throw new ArgumentNullException(nameof(localizedTextService));
+            _localizedTextService = localizedTextService ?? throw new ArgumentNullException(nameof(localizedTextService));
         }
 
+        //TODO: are there any repercussions elsewhere for us changing these to async?
+
         /// <summary>
         /// Gets the member group json for the member group id
         /// </summary>
         /// <param name="id"></param>
         /// <returns></returns>
-        public ActionResult<MemberGroupDisplay> GetById(int id)
+        public async Task<ActionResult<MemberGroupDisplay>> GetById(int id)
         {
-            var memberGroup = _memberGroupService.GetById(id);
+            //TODO: did we envisage this - combination of service and identity manager?
+            IdentityRole identityRole = await _roleManager.FindByIdAsync(id.ToString());
+            if (identityRole == null)
+            {
+                return NotFound();
+            }
+
+            IMemberGroup memberGroup = _memberGroupService.GetById(id);
             if (memberGroup == null)
             {
                 return NotFound();
             }
 
-            var dto = _umbracoMapper.Map<IMemberGroup, MemberGroupDisplay>(memberGroup);
+            //TODO: the default identity role doesn't have all the properties IMemberGroup had, e.g. CreatorId
+            MemberGroupDisplay dto = _umbracoMapper.Map<IMemberGroup, MemberGroupDisplay>(memberGroup);
             return dto;
         }
 
-
         /// <summary>
         /// Gets the member group json for the member group guid
         /// </summary>
         /// <param name="id"></param>
         /// <returns></returns>
-        public ActionResult<MemberGroupDisplay> GetById(Guid id)
+        public async Task<ActionResult<MemberGroupDisplay>> GetById(Guid id)
         {
-            var memberGroup = _memberGroupService.GetById(id);
+            //TODO: did we envisage just identity or a combination of service and identity manager?
```

Comment:
In these instances, not sure whether to replace with identity or remain as a service, because the display needs to map to the role anyway and we don't have all those properties

Reply:
We should only use our member services for these editors .The Role manager will almost solely be used for public access